### PR TITLE
vim: update version to 9.0.0181 to fix CVEs 2022-2522, 2022-2571, 2022-2580, 2022-2581.

### PR DESCRIPTION
**Pull request by PawelWMS**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [X] The toolchain has been rebuilt successfully (or no changes were made to it)
- [X] The toolchain/worker package manifests are up-to-date
- [X] Any updated packages successfully build (or no packages were changed)
- [X] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [X] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [X] All package sources are available
- [X] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [X] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [X] All source files have up-to-date hashes in the `*.signatures.json` files
- [X] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [X] Documentation has been updated to match any changes to the build system
- [X] Ready to merge

---

###### Summary 

Updating `vim` to the latest version to fix CVEs.

###### Change Log  

- Updated `vim` to version 9.0.0181.
- Sorted `cgmanifest.json` contents.

###### Does this affect the toolchain?  

No.

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2022-2522
- https://nvd.nist.gov/vuln/detail/CVE-2022-2571
- https://nvd.nist.gov/vuln/detail/CVE-2022-2580
- https://nvd.nist.gov/vuln/detail/CVE-2022-2581

###### Test Methodology

- Buddy build green.

### CODE REVIEW

File: cgmanifest.json [hunk 1]
```diff
@@ -2144,19 +2144,19 @@
       "component": {
         "type": "other",
         "other": {
-          "name": "hyperv-daemons",
```
Comment (anphel31) [inline cgmanifest.json:2147]: Can you check these cgmanifest.json changes? Looks like a number of unrelated packages have changed

**Comment by anphel31** [comment]
Can you update the CVE links in the description with "CVE-" before the numbers to fix the links

File: cgmanifest.json (hunk 1 above)
Comment (PawelWMS) [reply cgmanifest.json:2147]: This is correct. I've ran the script updating the manifest and it put packages in alphabetical order.

**Comment by PawelWMS** [comment]
> Can you update the CVE links in the description with "CVE-" before the numbers to fix the links

Done, thanks for catching that!

**Review by mfrw — APPROVED** [review]
LGTM

**Pull request merged**